### DISCUSSION TITLE
MAgPIE 4.0 - data download error

### DISCUSSION THREAD

**Discussion by jansteinhauser** [magpiemodel/magpie#335] [Q&A] [answered]
Hello again!
I want to do a few test runs on v4.0 to compare them to the runs released with the 4.0 paper in 2019. However, when I try downloading the required inputs, the script stops at `f12_interest_rate_bound.cs3` because:

`Error in write.magpie(round(read.magpie(input_file), round), output_file) :   Input data seems to be sparse but cs3 does not support sparse data. Please use cs4 instead!`

Any ideas what I can do about this? Could this be because some R packages are too new for this script?

Thanks and have a nice weekend!
Jan

**Comment by abhimishr** [comment, marked as answer]
I have a dirty fix for you, maybe @tscheypidi can suggest an elegant way of doing it because this concerns backward compatibility as well as reproducibility to a small extent.

1. Replace your `download_data_only.R` script with
```
require(lucode2)
require(gms)

source("scripts/start_functions.R")
source("config/default.cfg")
download_and_update(cfg)
```
2. Replace `start_functions.R` script with one of the most recent one (maybe from the master branch)

Execute download data step - will throw some errors but the model still compiles, maybe give it a try (hoping the run goes through).

Cheers,
Abhi

**Reply by jansteinhauser** [reply]
Hey, thanks for this. As a test, instead of step 1, I actually just added `require(gms)` to the `download_data_only.R` script and even that (plus step 2) was enough. Any specific reason for the other changes to the script?

**Reply by acsoterroni** [reply]
Hello,

I'm a new MagPIE user! To run version 4.0 I had to:
1) Replace `lucode` by `lucode2` everywhere (I used 'perl' command);
2) Add `require(gms)` to `./script/start/download_data_only.R` script;
3) Replace `./script/start_functions.R` script by the one from version 4.3.5.

I was able to generate the `fulldata.gdx` but my `default_validation.pdf` has only 3 pages! However, I was able to generate a complete PDF file for the version 4.3.5. I'm running MagPIE on Unix.

Best,
Aline

**Reply by abhimishr** [reply]
Welcome to MAgPIE @acsoterroni! 

Unfortunately, generation of validation PDF is bit more tricky to get right on local machines. See discussion from https://github.com/magpiemodel/magpie/discussions/333#discussioncomment-1583977 onwards. The log file in your model output folder might provide an error message on what went wrong. Please don't hesitate to open a new discussion thread about this, and then we can specifically have a look at it.

**Reply by abhimishr** [reply]
@jansteinhauser I just loaded up some library calls from a personal start script hence the calls to both `gms` and `lucode2`. Probably just calling `gms` is enough as you said!

cheers,
Abhi
